### PR TITLE
fix: adjust fill behavior of range query

### PR DESCRIPTION
**Pull request by Taylor-lagrange**
I hereby agree to the terms of the [GreptimeDB CLA](https://github.com/GreptimeTeam/.github/blob/main/CLA.md).

## What's changed and what's your intention?

Adjust fill behavior on range query

- When user not use `fill`, the null result calculate by range expr, or the missing datapoint will not be fill.
- If user use fill `NULL`, the missing datapoint will be fill.
- If user use fill `LINEAR/PREV/Const`, all null value and missing datapoint will be fill.

Case like

```sql
CREATE TABLE host (
  ts timestamp(3) time index,
  host STRING PRIMARY KEY,
  val BIGINT,
);

INSERT INTO TABLE host VALUES
    (0,     'host1', 0),
    -- missing data for 5000, 10000
    (15000, 'host1', 6),
    (0,     'host2', 6),
    -- missing data for 5000, 10000
    (15000, 'host2', 12);

SELECT ts, host, min(val) RANGE '5s' FROM host ALIGN '5s' ORDER BY host, ts;

+---------------------+-------+------------------------+
| ts                  | host  | MIN(host.val) RANGE 5s |
+---------------------+-------+------------------------+
| 1970-01-01T00:00:00 | host1 | 0                      |
| 1970-01-01T00:00:15 | host1 | 6                      |
| 1970-01-01T00:00:00 | host2 | 6                      |
| 1970-01-01T00:00:15 | host2 | 12                     |
+---------------------+-------+------------------------+

SELECT ts, host, min(val) RANGE '5s' FILL NULL FROM host ALIGN '5s' ORDER BY host, ts;

+---------------------+-------+----------------------------------+
| ts                  | host  | MIN(host.val) RANGE 5s FILL NULL |
+---------------------+-------+----------------------------------+
| 1970-01-01T00:00:00 | host1 | 0                                |
| 1970-01-01T00:00:05 | host1 |                                  |
| 1970-01-01T00:00:10 | host1 |                                  |
| 1970-01-01T00:00:15 | host1 | 6                                |
| 1970-01-01T00:00:00 | host2 | 6                                |
| 1970-01-01T00:00:05 | host2 |                                  |
| 1970-01-01T00:00:10 | host2 |                                  |
| 1970-01-01T00:00:15 | host2 | 12                               |
+---------------------+-------+----------------------------------+

SELECT ts, host, min(val) RANGE '5s' FILL PREV FROM host ALIGN '5s' ORDER BY host, ts;

+---------------------+-------+----------------------------------+
| ts                  | host  | MIN(host.val) RANGE 5s FILL PREV |
+---------------------+-------+----------------------------------+
| 1970-01-01T00:00:00 | host1 | 0                                |
| 1970-01-01T00:00:05 | host1 | 0                                |
| 1970-01-01T00:00:10 | host1 | 0                                |
| 1970-01-01T00:00:15 | host1 | 6                                |
| 1970-01-01T00:00:00 | host2 | 6                                |
| 1970-01-01T00:00:05 | host2 | 6                                |
| 1970-01-01T00:00:10 | host2 | 6                                |
| 1970-01-01T00:00:15 | host2 | 12                               |
+---------------------+-------+----------------------------------+
```

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.
- [ ]  This PR does not require documentation updates.

### CODE REVIEW

**Comment by killme2008** [comment]
@tisonkun Is this behavior you expected?

File: src/query/src/range_select/plan.rs [hunk 1]
```diff
@@ -1141,29 +1152,47 @@ impl RangeSelectStream {
...
         {
-            // collect data on time series
-            let mut align_ts = align_ts_accumulator.keys().copied().collect::<Vec<_>>();
-            if need_sort_output {
-                align_ts.sort();
+            // skip empty time series
+            if align_ts_accumulator.is_empty() {
+                continue;
             }
+            // find the first and last align_ts
```
Comment (killme2008) [inline src/query/src/range_select/plan.rs:1177]: Can we add a fast path when `align_ts_accumulator` only has one element?

**Review by killme2008 — APPROVED** [review]
LGTM

File: src/query/src/range_select/plan.rs (hunk 1 above)
Comment (Taylor-lagrange) [reply src/query/src/range_select/plan.rs:1177]: In this scenario, many dispose logic still duplicate, and the readability of the code will decrease.

**Review by tisonkun — APPROVED** [review]
+1

Update PR description we need docs update for this.

cc @nicecui

**Pull request merged**